packages/multichain/src/handlers/wallet-invokeMethod.ts

```diff
@@ -62,10 +62,9 @@ async function walletInvokeMethodHandler(
     return end(providerErrors.unauthorized());
   }
 
-  const scopeObject = mergeScopes(
-    caveat.value.requiredScopes,
-    caveat.value.optionalScopes,
-  )[scope];
+  const scopeObject = getSessionScopes(caveat.value)[
+    scope as InternalScopeString
```

Comment:
This suggests to me that we need some type validation above to assert that the `scope` is not a `namespace` (other than `wallet`)

Reply:
https://github.com/MetaMask/core/pull/4961/commits/e7373ae5d068381ca32239c2accd45276aeda745